src/member.rs

```diff
@@ -653,6 +681,10 @@ where
                 // TODO: these responses are quite heavy, we should at some point add
                 // checks to make sure we are not processing responses to request we did not make.
                 // TODO: we need to check if the response (and alert) does not exceed some max message size in network.
+                let parents = parents
+                    .into_iter()
+                    .filter_map(|unchecked| Signed::from(unchecked, self.keybox))
```

Comment:
In the context of Michał's comments above we should probably also note the wrong signatures here somehow, although it's much more annoying.

Reply:
Turns out that `Result<Vec<T>, E>` [implements](https://doc.rust-lang.org/std/result/enum.Result.html#impl-FromIterator%3CResult%3CA%2C%20E%3E%3E) `FromIterator<Result<T, E>>`, which gives a quite nice solution to catch an error if there is any.